projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -1175,6 +1175,16 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
     @ViewChild('defaultEmptyGrid', { read: TemplateRef, static: true })
     public emptyGridDefaultTemplate: TemplateRef<any>;
 
+    /**
+     * A template reference for the template if all rows are selected.
+     *
+     * @example
+     * ```
+     * const areAllRowsSelected = this.grid.isSelectAllClicked;
+     * ```
+     */
+    public isSelectAllClicked = false;
```

Comment:
There shouldn't be a property inside the grid for this. This is an event driven flag and it should exist only in the lifecycle of the interaction raising the event.

Also, `isSelectAllClicked` is not an appropriate name for the flag. There could be other interactions, not only `click` that cause all rows to be selected.

Reply:
I didn't quite understand what you mean. What defines " the lifecycle of the interaction raising the event."? 

The issue to solve -  subscribe on the `dataPreload` event - we need to know if the Select all checkbox is/was selected. Where would this flag live, if not on the grid?

Reply:
The issue described doesn't subscribe to the `dataPreload`, instead it subscribes to the `rowSelectionChanging` and tries to determine whether any data loaded in the future should also be selected, or not. The application scenario can use the flag from the `rowSelectionChanging` event to populate a component-level property with a flag of whether newly loaded data is to be selected, or not, but that shouldn't be done on a library level by populating a flag belonging to the grid itself.

Reply:
We went ahead to see if implementing this scenario on library level would be ok. In the end, it seemed possible even without populating a flag in the grid itself.

Can evaluate the resolution after @Svetloslav15  make the final commits/

Reply:
I've removed the flag from the component and now if the user wants to automatically select remote data when scrolling, they should subscribe to rowSelectionChanging and dataChanged event, and handle selection on application level.